python/src/linalg.cpp

```diff
@@ -353,4 +353,41 @@ void init_linalg(nb::module_& parent_module) {
         Returns:
           array: :math:`A^{-1}` where :math:`\mathbf{A} = \mathbf{L}\mathbf{L}^T`.
       )pbdoc");
+  m.def(
+      "eigvalsh",
+      &eigvalsh,
+      "a"_a,
+      "upper"_a = true,
+      nb::kw_only(),
+      "stream"_a = nb::none(),
+      nb::sig(
+          "def eigvalsh(a: array, upper: bool = True, *, stream: Union[None, Stream, Device] = None) -> array"),
+      R"pbdoc(
+        Compute the eigenvalues of a complex Hermitian or real symmetric matrix.
```

Comment:
We don't do a brilliant job of this in the other `linalg` functions, but we should probably check if complex inputs work and raise a not implemented error if not.